token/cli/src/clap_app.rs

```diff
@@ -693,6 +693,13 @@ pub fn app<'a, 'b>(
                             "Enable the mint authority to close this mint"
                         ),
                 )
+                .arg(
+                    Arg::with_name("token_program_2022")
+                        .value_name("token-program-2022")
+                        .short("2")
+                        .long("token-program-2022")
+                        .takes_value(false)
+                        .help("Use token extension program token 2022 with program id: TokenzQdBNbLqP5VEhdkAS6EPFLC1PHnBqCXEpPxuEb"))
```

Comment:
This is in the wrong spot. The `--program-id` option lives under `app`:

https://github.com/solana-labs/solana-program-library/blob/d2d1fcd4402a8179fa57efbcba34f3273dfb9ce8/token/cli/src/clap_app.rs#L594-L603

Let's move it over there so it can apply to other commands besides just `create-token`, and be sure to use `.conflicts_with` to make sure someone can't provide this arg and `--program-id` together.

Also, what do you think about just `--program-2022`? Nice & short.

Reply:
Updated :) I added it only to the create-token because all others resolve the id automatically. But makes sense to have it global nevertheless :)